app/src/main/java/com/example/android/uamp/fragments/SettingsScreen.kt

```diff
@@ -0,0 +1,209 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.uamp.fragments
+
+import android.media.AudioAttributes
+import android.media.AudioFormat
+import android.media.AudioManager
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Switch
+import androidx.compose.material.SwitchDefaults
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
+import androidx.lifecycle.viewModelScope
+import androidx.media3.common.C
+import androidx.navigation.NavController
+import com.example.android.uamp.R
+import com.example.android.uamp.viewmodels.MainActivityViewModel
+import kotlinx.coroutines.launch
+
+
+/**
+ * SettingsScreenDescription serves to describe the UI of the settings screen, which includes a TopAppBar
+ * for the user to return to the NowPlayingScreen, a switch to enable spatial audio, and spatializer
+ * function outputs of the current mediaItem being played.
+ *
+ * @param mainActivityViewModel to reference functions in MainActivityViewModel
+ * @param navController to allow navigation back to NowPlayingScreen
+ */
+@Composable
+fun SettingsScreenDescription(
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val mCheckedValue = remember { mutableStateOf(false) }
+    val audioManager =
+        ContextCompat.getSystemService(LocalContext.current, AudioManager::class.java)
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
```

Comment:
Just curious, is it possible to have the Settings screen only take up the height it needs to display its content (i.e. the equivalent of `wrap_content` in XML) and let the Now Playing screen peek through underneath?

Reply:
Under NavHost, navigating to a new screen by default results in the screen taking up the entire screen. This would be possible without using NavHost but will require some changes in how the navigation logic is implemented.